src/lib.rs

```diff
@@ -469,6 +511,14 @@ impl<'a, 'input> Attribute<'a, 'input> {
         &self.data.value
     }
 
+    /// Returns attribute's shared value.
+    ///
+    /// Unlike a string returned by `value()`, `SharedString` can outlive the [`Document`].
+    #[inline]
+    pub fn shared_value(&self) -> SharedString<'input> {
```

Comment:
I think it would be more flexible to expose `&SharedString<'input>` and let the caller clone if necessary.

Reply:
Not applicable if we go for on-demand sharing as detailed below.

Reply:
But probably a requirement if this implemented for e.g. text nodes as we other need to duplicate the implementation of `fn text(&self) -> ...`. With `fn shared_text(&self) -> &SharedString<'input>` we could implement `fn text(&self) -> &str` in terms of that.

Reply:
As always, it wasn't as easy. We store comment as a string and return it via `Node::text`. Meaning we cannot return `&SharedString` here.

We should either store comment as `SharedString` as well or have a separate method for getting the comment. Which is what I did. This is a breaking change, but a minor one.